app/src/main/java/net/gsantner/markor/format/todotxt/TodoTxtTask.java

```diff
@@ -209,6 +212,19 @@ public String getDueDate(final String defaultValue) {
         return dueDate;
     }
 
+    public TodoDueState getDueStatus() {
+        if (dueStatus == null) {
+            final String date = getDueDate();
+            if (TextUtils.isEmpty(date)) {
+                dueStatus = TodoDueState.NONE;
+            } else {
+                final int comp = date.compareTo(getToday());
+                dueStatus = (comp > 0) ? TodoDueState.FUTURE : (comp < 0) ? TodoDueState.OVERDUE : TodoDueState.TODAY;
```

Comment:
For future "this week" and "this month" might be interesting.

not for this PR.